core/src/main/scala/filodb.core/store/ChunkSource.scala

```diff
@@ -146,7 +146,7 @@ trait DefaultChunkSource extends ChunkSource {
       // NOTE: this executes the partMaker single threaded.  Needed for now due to concurrency constraints.
       // In the future optimize this if needed.
       .mapAsync { rawPart => partMaker(dataset, partMethod.shard).populateRawChunks(rawPart)
-                               .executeOn(singleThreadPool) }
+                               .executeOn(singleThreadPool).asyncBoundary }
```

Comment:
Same here.  Evaluate whether populateRawChunks really needs to be on the singleThreadPool

Reply:
@velvia It is needed. The locks take care of chunk-map operations.
Partition creation modifies the shard state, namely PartitionSet, PartitionMap etc which need to be altered by ingestion thread only (unless we add additional locking).